### PR TITLE
Update gitignore decorations when .git/info/exclude file is edited

### PR DESCRIPTION
**Pull request by vpukhanov**
[According to git-scm](https://git-scm.com/docs/gitignore),

> Patterns which are specific to a particular repository but which do not need to be shared with other related repositories (e.g., auxiliary files that live inside the repository but are specific to one user’s workflow) should go into the `$GIT_DIR/info/exclude` file.

I use this file from time to time to remove some personal workflow specific files from the repo. Currently, when this file is edited, the file tree decorations do not update (the ignored files do not "gray out"), requiring me to reopen the repo for the update to occur. This PR fixes this issue.

### CODE REVIEW

File: extensions/git/src/decorationProvider.ts [hunk 1]
```diff
@@ -78,6 +78,13 @@ class GitIgnoreDecorationProvider implements DecorationProvider {
 		}
 	}
 
+	private isIgnoreConfig(document: TextDocument): boolean {
+		return (
+			document.fileName.endsWith('.gitignore') ||
+			document.fileName.replace(/\\/g, '/').endsWith('.git/info/exclude')
```
Comment (joaomoreno) [inline extensions/git/src/decorationProvider.ts:84]: Are you sure `fileName` includes the full path and is not just the file name?

**Review by joaomoreno — CHANGES_REQUESTED** [review]
Could we instead just have a regex here?

File: extensions/git/src/decorationProvider.ts (hunk 1 above)
Comment (vpukhanov) [reply extensions/git/src/decorationProvider.ts:84]: Yes, it's a shorthand for [uri.fsPath](https://github.com/microsoft/vscode/blob/c73b3ca474af58e0ceaf827d5d5bd698b338fe52/src/vs/vscode.d.ts#L105).

**Comment by vpukhanov** [comment]
@joaomoreno regex makes sense. I don't really love the way we have to match the system-dependent path separators. What do you think?

    \.gitignore$|\.git[\/\\]info[\/\\]exclude$

**Comment by vpukhanov** [comment]
I guess we can use [uri.path](https://github.com/microsoft/vscode/blob/c73b3ca474af58e0ceaf827d5d5bd698b338fe52/src/vs/vscode.d.ts#L1351) instead, because it always uses forward slash as a separator. Will submit a change.

**Review by joaomoreno — APPROVED** [review]
👏

**Pull request merged**